src/bors/handlers/review.rs

```diff
@@ -81,6 +83,10 @@ async fn check_pr_approval_validity(
         return Ok(Some(approve_non_open_pr_comment()));
     }
 
+    if matches!(pr.github.mergeable_state, MergeableState::Dirty) {
```

Comment:
We do indeed treat both `Dirty` and `Blocked` as having conflicts. Could you please construct `database::MergeableState`  from the GH state using its `From` impl, and then check if the resulting state is mergeable? To use the same logic everywhere. Thanks!